src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -141,22 +136,48 @@ public CourseResource(Environment env, UserService userService, CourseService co
         if (course.getId() != null) {
             throw new BadRequestAlertException("A new course cannot already have an ID", ENTITY_NAME, "idexists");
         }
+
+        // Check if course shortname matches regex
+        Matcher shortNameMatcher = SHORT_NAME_PATTERN.matcher(course.getShortName());
+        if (!shortNameMatcher.matches()) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The shortname is invalid", "shortnameInvalid")).body(null);
+        }
+
+        List<Course> coursesWithSameShortName = courseRepository.findAllByShortName(course.getShortName());
+        if (coursesWithSameShortName.size() > 0) {
+            return ResponseEntity.badRequest().headers(
+                    HeaderUtil.createAlert(applicationName, "A course with the same short name already exists. Please choose a different short name.", "shortnameAlreadyExists"))
+                    .body(null);
+        }
+
+        if (course.getStudentGroupName() != null) {
+            throw new BadRequestAlertException("The student group name must be null when creating a course", ENTITY_NAME, "studentGroupNameCannotBeSet");
+        }
+        if (course.getTeachingAssistantGroupName() != null) {
+            throw new BadRequestAlertException("The teaching assistant group name must be null when creating a course", ENTITY_NAME, "teachingAssistantGroupNameCannotBeSet");
+        }
+        if (course.getInstructorGroupName() != null) {
+            throw new BadRequestAlertException("The instructor group name must be null when creating a course", ENTITY_NAME, "instructorGroupNameCannotBeSet");
+        }
+
         validateComplaintsConfig(course);
+
+        course.setStudentGroupName("artemis-" + course.getShortName() + "-students");
+        course.setTeachingAssistantGroupName("artemis-" + course.getShortName() + "-tutors");
+        course.setInstructorGroupName("artemis-" + course.getShortName() + "-instructors");
```

Comment:
I am wondering if we should create the group names in the Course Object, as we use them multiple times.

Reply:
good point. I refactored it